BiweeklyMeetings/2022-09-06.md

```diff
@@ -0,0 +1,353 @@
+
+**Matthias** -: Maybe there is a CIP to make here to suggest the removal of pointer addresses, since I think we are amongst the two individuals that have been using them on the chain, the only two. So yeah, that will simplify a bit the things. Okay. Should we get that CIP for review next week? Is the discussion still ongoing? Or I see mostly happened months ago, but no, no-
+
+**Sebastien** -: No, I don't think there's any ongoing discussion anymore. I think everybody has said their thoughts on the proposed path. I think some of the stuff has not been decided for the CIP 0039, the language annotated addresses. Michael Peyton Jones and Jared wanted to try and figure out how we can make this an on-chain construct that is actually validated by the blockchain itself. But then you need some way for the blockchain to know ahead of time whether not an address is a native script or a Plutus script. Or the other option is that if you don't validate this ahead of time, but then if somebody sends to the wrong script type, they can never spend from it, which is also not great.
+
+**Matthias** -: So it could be done fully optionally. So if you specify a type and you provide the script part of the transaction, then the ledger validates it. And if it doesn't validate, then it fails. So as a client, you have a way to sort of enforce that there is a ledger validation on top, but if you don't provide it, then yes, the ledger has no way to validate it. Then it's just blindly accepts it, like it does now.
+
+**Sebastien** -: Yeah. And the question is whether not this validation should have been when you send to the address or when you spend from the address.
+**Matthias** -: Well, I would say if you can do it when you send to the address, that's where the interest is at this stage, because that's where I think it can happen right now.
+
+**Sebastien** -: Yeah. That was my preference as well. Michael Peyton Jone's preferred to do the validation when you spend from the address.
+
+**Matthias** -: When you spend from it, you can always do it, because you have the script at this stage. But then what happens if you have sent to an address which has a mismatch between the type and the actual script, you are just locked forever.
+
+**Sebastien** -: Yeah. That's Michael Peyton Jone's suggestion.
```

Comment:
```suggestion
**Sebastien** -: Yeah. That's Michael Peyton Jones's suggestion.
```
https://www.merriam-webster.com/words-at-play/what-happens-to-names-when-we-make-them-plural-or-possessive